### PR TITLE
Update HTMLCollection and Add HTMLCollectionOf<T extends Element>

### PR DESCRIPTION
**Pull request by FranklinWhale**
First PR for [Microsoft/TypeScript#424](https://github.com/Microsoft/TypeScript/issues/424)

### CODE REVIEW

File: baselines/dom.generated.d.ts [hunk 1]
```diff
@@ -7047,7 +7047,7 @@ interface HTMLTableElement extends HTMLElement {
     /**
       * Retrieves a collection of all tBody objects in the table. Objects in this collection are in source order.
       */
-    tBodies: HTMLCollection;
+    tBodies: HTMLCollectionOf<HTMLTableSectionElement>;
```
Comment (zhengbli) [inline baselines/dom.generated.d.ts]: Why not `HTMLTableBodyElement`?

Comment (FranklinWhale) [inline baselines/dom.generated.d.ts]: I followed the return type of the following functions:
```ts
createElement(tagName: "tbody"): HTMLTableSectionElement;
getElementsByTagName(tagname: "tbody"): NodeListOf<HTMLTableSectionElement>;
createTBody(): HTMLTableSectionElement;
```

File: inputfiles/overridingTypes.json [hunk 2]
```diff
@@ -267,18 +333,42 @@
...
         "kind": "method",
         "interface": "HTMLTableRowElement",
         "name": "insertCell",
         "signatures": ["insertCell(index?: number): HTMLTableCellElement"]
     },
     {
+        "kind": "property",
+        "interface": "HTMLTableRowElement",
+        "name": "cells",
+        "type": "HTMLCollectionOf<HTMLTableCellElement>"
```
Comment (mhegazy) [inline inputfiles/overridingTypes.json]: as per https://html.spec.whatwg.org/multipage/tables.html#dom-tr-cells, this should be `HTMLTableCellElement | HTMLTableHeaderCellElement`

**Comment by mhegazy** [comment]
:+1:

File: inputfiles/overridingTypes.json (hunk 2 above)
Comment (FranklinWhale) [inline inputfiles/overridingTypes.json]: `HTMLTableCellElement` is the super interface of `HTMLTableDataCellElement` and `HTMLTableHeaderCellElement`. Should I change it to `HTMLTableDataCellElement | HTMLTableHeaderCellElement`?

Comment (mhegazy) [inline inputfiles/overridingTypes.json]: the result should be the same, given the current definitions. but for accuracy i would change it.

Comment (FranklinWhale) [inline inputfiles/overridingTypes.json]: Changed. Also changed tr.insertCell to return HTMLTableDataCellElement.

**Comment by mhegazy** [comment]
Thanks. LGTM.

**Pull request merged**